controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -91,6 +92,7 @@ func (a *Adapter) EnsureIntegrationTestPipelineForScenarioExists() (controller.O
 	a.logger.Info("The SnapshotEnvironmentBinding's deployment succeeded", "snapshotEnvironmentBinding.Name", a.snapshotEnvironmentBinding.Name)
 
 	gitops.PrepareAndRegisterSEBReady(a.snapshotEnvironmentBinding)
+	metrics.RegisterSEBSuccessfulDeployment()
```

Comment:
Any RequeueWithError below will cause false positive increase in number of successful deployments

Reply:
Moved after those RequeueWithError sections.

Reply:
yes, but it's not a complete fix, its more like workaround. Any extra reconcilaition loops will cause false positive increase, also `metrics.RegisterSEBSuccessfulDeployment()` should be moved at the end

Reply:
Moved before: `return controller.ContinueProcessing()`